### PR TITLE
HardforkExclusionHandler into header interceptor

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- integrate HardforkExclusionHandler into interceptor in order to not validate headers from excluded intervals
  
## Proposed changes
- integrated HardforkExclusionHandler

## Testing procedure
- will be tested on feat branch

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: integrationTests/factory/consensusComponents/consensusComponents_test.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ func TestConsensusComponents_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
-	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents)
+	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents, &testscommon.HardforkExclusionHandlerStub{})
```
Comment (iulianpascalau) [inline integrationTests/factory/consensusComponents/consensusComponents_test.go:44]: multiple lines?

File: integrationTests/factory/dataComponents/dataComponents_test.go [hunk 2]
```diff
@@ -33,7 +34,7 @@ func TestDataComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
-	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents)
+	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents, &testscommon.HardforkExclusionHandlerStub{})
```
Comment (iulianpascalau) [inline integrationTests/factory/dataComponents/dataComponents_test.go:37]: multiple lines?

File: integrationTests/factory/heartbeatComponents/heartbeatComponents_test.go [hunk 3]
```diff
@@ -40,7 +41,7 @@ func TestHeartbeatComponents_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
-	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents)
+	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents, &testscommon.HardforkExclusionHandlerStub{})
```
Comment (iulianpascalau) [inline integrationTests/factory/heartbeatComponents/heartbeatComponents_test.go:44]: multiple lines?

File: integrationTests/factory/stateComponents/stateComponents_test.go [hunk 4]
```diff
@@ -37,7 +38,7 @@ func TestStateComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
-	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents)
+	managedBootstrapComponents, err := nr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents, &testscommon.HardforkExclusionHandlerStub{})
```
Comment (iulianpascalau) [inline integrationTests/factory/stateComponents/stateComponents_test.go:41]: multiple lines?

**Review by iulianpascalau — COMMENTED** [review]
GJ :+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**